fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergPartitionUtils.java

```diff
@@ -131,4 +144,87 @@ public static Set<IcebergPartition> getAllPartition(Table table) {
+    // month(ts) partitionName : 2023-01           return 2023-01-01 (Date) or 2023-01-01 00:00:00 (Datetime)
+    // day(ts)   partitionName : 2023-01-01        return 2023-01-01 (Date) or 2023-01-01 00:00:00 (Datetime)
+    // hour(ts)  partitionName : 2023-01-01-12     return 2023-01-01 12:00:00 (Datetime)
+    public static String normalizeTimePartitionName(String partitionName, PartitionField partitionField, Schema schema,
+                                                    Type type) {
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        boolean parseFromDate = true;
+        if (partitionField.transform().toString().equalsIgnoreCase("year")) {
+            partitionName += "-01-01";
+        } else if (partitionField.transform().toString().equalsIgnoreCase("month")) {
+            partitionName += "-01";
+        } else if (partitionField.transform().toString().equalsIgnoreCase("day")) {
+            dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        } else if (partitionField.transform().toString().equalsIgnoreCase("hour")) {
+            dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd-HH");
+            parseFromDate = false;
+        }
+
+        // partition name formatter
+        DateTimeFormatter formatter = null;
+        if (type.isDate()) {
+            formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        } else {
+            formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+        }
+        // If has timestamp with time zone, should compute the time zone offset to UTC
+        ZoneId zoneId;
+        if (schema.findType(partitionField.sourceId()).equals(Types.TimestampType.withZone())) {
+            zoneId = TimeUtils.getTimeZone().toZoneId();
+        } else {
+            zoneId = ZoneOffset.UTC;
+        }
+
+        String result;
+        try {
+            if (parseFromDate) {
+                LocalDate date = LocalDate.parse(partitionName, dateTimeFormatter);
+                if (type.isDate()) {
+                    result = date.format(formatter);
+                } else {
+                    LocalDateTime dateTime = date.atStartOfDay().atZone(ZoneOffset.UTC).
+                            withZoneSameInstant(zoneId).toLocalDateTime();
+                    result = dateTime.format(formatter);
+                }
+            } else {
+                // parse from datetime which contains hour
+                LocalDateTime dateTime = LocalDateTime.parse(partitionName, dateTimeFormatter).atZone(ZoneOffset.UTC).
+                        withZoneSameInstant(zoneId).toLocalDateTime();
+                result = dateTime.format(formatter);
+            }
+        } catch (Exception e) {
+            LOG.warn("parse partition name failed, partitionName: {}, partitionField: {}, type: {}",
+                    partitionName, partitionField, type);
+            result = partitionName;
```

Comment:
is it reasonable to swallow the exception, why not throw again?

Reply:
Fixed